provider/etcd/etcd_client.go

```diff
@@ -323,39 +303,47 @@ func (ec *EtcdClient) RunLoadAllAlerts(ctx context.Context) {
 					continue // retry
 				}
 				count += 1
-				_ = ec.alerts.Put(alert) // best effort only
+				_ = ec.alerts.PutFromEtcd(alert) // best effort only
 			}
 			level.Info(ec.logger).Log("msg", "Etcd Load All Alerts Finished", "count", count)
 			return // we only need to load all of the alerts once
 		}
 	}()
 }
 
-// Equals returns a true if the two alerts have the same values, false otherwise
-func AlertsEqualExceptForUpdatedAt(a *types.Alert, o *types.Alert) bool {
+func AlertsShouldWriteToEtcd(a *types.Alert, o *types.Alert) bool {
+	// Check if the alerts are "different" enough.
+	// If alerts ARE "different" enough then return 'true' in order to write to Etcd
+	// If alerts are NOT "different" enough then return 'false' to skip writing to etcd
+
 	if a == nil || o == nil {
-		return false
+		return true
 	}
 	if !reflect.DeepEqual(a.Labels, o.Labels) {
-		return false
+		return true
 	}
 	if !reflect.DeepEqual(a.Annotations, o.Annotations) {
-		return false
+		return true
 	}
 	if a.GeneratorURL != o.GeneratorURL {
-		return false
+		return true
 	}
 	if !a.StartsAt.Equal(o.StartsAt) {
-		return false
+		return true
 	}
-	if !a.EndsAt.Equal(o.EndsAt) {
-		return false
+
+	// Write to etcd if EndsAt's are "different" enough
+	significantTimeDifference := 300 * time.Second
+	if (a.EndsAt.Before(o.EndsAt) && o.EndsAt.Sub(a.EndsAt) > significantTimeDifference) || (o.EndsAt.Before(a.EndsAt) && a.EndsAt.Sub(o.EndsAt) > significantTimeDifference) {
+		// Update because EndsAt is different enough
+		return true
```

Comment:
How much does this save if we are now doing PutFromEtcd()? Just a guess, but I think it would only prevent repeat alerts from POSTs from being updated, which is just a single write per-alert for every cycle (~90s for our aci scrapes). And it lets am's drift just a little bit.

Reply:
@mpeters You're right.  Perhaps we need to remove this bit of code.  What do you think?

Reply:
We should test it, but I'm in favor of removing since it's seems a bit janky (heuristic that might  be wrong sometimes and lead to drift).

Reply:
@mpeters I'll remove it.  Great recommendation.